### ISSUE TITLE
Support for @sentry/vue v8

### ISSUE THREAD

**Issue by pkissling** [supabase-community/sentry-integration-js#9]
# Feature request

## Is your feature request related to a problem? Please describe.

I just wanted to update my Vue 3 application to use [@sentry/vue v8](https://github.com/getsentry/sentry-javascript/releases/tag/8.0.0). This fails due to type issues of `SupabaseIntegration`:

```
Type 'SupabaseIntegration' is not assignable to type 'Integration'.
  Types of property 'setupOnce' are incompatible.
    Type '(addGlobalEventProcessor: (callback: EventProcessor) => void, getCurrentHub: () => Hub) => void' is not assignable to type '() => void'.
      Target signature provides too few arguments. Expected 2 or more, but got 0.
```

This, I think, i due to the [removal of class based integrations in favor of function based ones](https://github.com/getsentry/sentry-javascript/blob/8.0.0/MIGRATION.md#removal-of-class-based-integrations):

## Describe the solution you'd like

Implement the function based integrations approach for `sentry-integration-js`

**Comment by KirioXX** [comment]
Same error message for `@sentry/react v8`

**Comment by kbsali** [comment]
please!!! :pray:

**Comment by kamilogorek** [comment]
`0.2.0-beta` has been published. Give it a go and let me know if everything works.
Once tested, I'll mark it as latest and release a non-beta version. Thanks!

**Comment by kamilogorek** [comment]
Just a note that you'll need to use `supabaseIntegration` function and pass `Sentry` object to the integration itself now (see updated readme for usage).

**Comment by pkissling** [comment]
> `0.2.0-beta` has been published. Give it a go and let me know if everything works. Once tested, I'll mark it as latest and release a non-beta version. Thanks!

hi @kamilogorek, thank you for looking into this.
i did the upgrade and the migration using the following versions:
```
{
    "@sentry/vue": "^8.0.0",
    "@supabase/sentry-js-integration": "^0.2.0-beta"
}
```
But when i build the application, i get an error realted to version `v7.js` not being available:
```
RollupError: Could not resolve "./v7.js" from "node_modules/@supabase/sentry-js-integration/index.js"
file: ${SOME_DIR}/node_modules/@supabase/sentry-js-integration/index.js
    at getRollupError (file://${SOME_DIR}/node_modules/rollup/dist/es/shared/parseAst.js:376:41)
    at error (file://${SOME_DIR}/node_modules/rollup/dist/es/shared/parseAst.js:372:42)
    at ModuleLoader.handleInvalidResolvedId (file://${SOME_DIR}/node_modules/rollup/dist/es/shared/node-entry.js:18891:24)
    at file://${SOME_DIR}/node_modules/rollup/dist/es/shared/node-entry.js:18851:26
ERROR: "build-only" exited with 1.
```
am i doing something wrong?

**Comment by kamilogorek** [comment]
Ah, my bad. `0.2.0-beta.2` with the fix released. Sorry about that :)

**Comment by Keagel** [comment]
For what it's worth I get a similar error using Sentry for SvelteKit and the `0.2.0-beta.2` version:

```json
"dependencies": {
    "@sentry/sveltekit": "^8.2.1",
    "@supabase/sentry-js-integration": "^0.2.0-beta.2",
}
```

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'node_modules/@supabase/sentry-js-integration/common' imported from node_modules/@supabase/sentry-js-integration/v7.js
    at finalizeResolution (node:internal/modules/esm/resolve:264:11)
    at moduleResolve (node:internal/modules/esm/resolve:924:10)
    at defaultResolve (node:internal/modules/esm/resolve:1148:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:390:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:359:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:234:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)
    at link (node:internal/modules/esm/module_job:86:36) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file://node_modules/@supabase/sentry-js-integration/common'
```

Why is v7 being imported here?

I just tried using `0.2.0-beta.2` with a v7 of Sentry and that same error happens, so it's not just about using v8.

**Comment by pkissling** [comment]
> Ah, my bad. `0.2.0-beta.2` with the fix released. Sorry about that :)

with `0.2.0-beta.2` i am able to build and deploy the application, but the application is not reporting database queries to sentry anymore.

**Comment by kamilogorek** [comment]
@pkissling do you have any repro for this? Did you try to use debug build of Sentry SDK or `beforeSend` hook to see if anything is going outside of your app?
I have integration tests that make sure that events will be enhanced with the database traces, but sending mechanism itself is on the SDK responsibility side.

**Comment by pkissling** [comment]
> @pkissling do you have any repro for this? Did you try to use debug build of Sentry SDK or `beforeSend` hook to see if anything is going outside of your app? I have integration tests that make sure that events will be enhanced with the database traces, but sending mechanism itself is on the SDK responsibility side.

sorry, my bad. i did not configure my app properly. everything works fine for me!

**Comment by kamilogorek** [comment]
Released non-beta version of `0.2.0` to npm and github releases.

**Comment by pkissling** [comment]
> Released non-beta version of `0.2.0` to npm and github releases.

thank you!
